packages/circuit-ui/components/NotificationToast/NotificationToast.tsx

```diff
@@ -0,0 +1,218 @@
+/**
+ * Copyright 2021, SumUp Ltd.
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { HTMLAttributes, RefObject, useEffect, useRef, useState } from 'react';
+import { css } from '@emotion/react';
+import { Alert, Confirm, Info, Notify } from '@sumup/icons';
+
+import styled, { StyleProps } from '../../styles/styled';
+import { useAnimation } from '../../hooks/useAnimation';
+import Body from '../Body';
+import { TrackingProps } from '../../hooks/useClickEvent';
+import CloseButton from '../CloseButton';
+import { ClickEvent } from '../../types/events';
+import { BaseToastProps, createUseToast } from '../ToastContext';
+import { hideVisually } from '../../styles/style-mixins';
+
+const TRANSITION_DURATION = 200;
+const DEFAULT_HEIGHT = 'auto';
+
+type Variant = 'info' | 'confirm' | 'notify' | 'alert';
+
+export type NotificationToastProps = HTMLAttributes<HTMLDivElement> &
+  BaseToastProps & {
+    /**
+     * Notification toast variants. Defaults to `info`.
+     */
+    variant?: Variant;
+    /**
+     * Notification toast headline to provide information (optional)
+     */
+    headline?: string;
+    /**
+     * A body copy to provide notification toast information
+     */
+    body: string;
+    /**
+     * Whether the notification toast is visible.
+     */
+    isVisible: boolean;
+    /**
+     * Additional data that is dispatched with the tracking event.
+     */
+    tracking?: TrackingProps;
+    /**
+     * Close button is always rendered in the top right corner
+     * onClose is called when the toast is auto-dismissed or the close button is clicked.
+     */
+    onClose?: (event: ClickEvent) => void;
+    /**
+     * A clear and concise description of the icon and the Toast's purpose.
+     */
+    iconLabel: string;
```

Comment:
We decided that this should be required, right? I forgot.

I think it makes sense (ensures that developers think about it), but we should probably make the prop description clearer then, something like "Pass an empty string if the toast body is self-explanatory"

Reply:
We did, and I agree it will ensure that developers will think about it. I updated the description.